### DISCUSSION TITLE
Issue with Teleport in k8s using Cloudflare Origin certificates for domain

### DISCUSSION THREAD

**Discussion by dudeisbrendan03** [gravitational/teleport#8462] [Q&A]
Hi there,
We're using Cloudflare Origin and Teleport in Kubernetes and we're experiencing an issue building a train of trust for Teleport, it's saying that there is an issue building trust although we've provided Teleport with the issuer of the certificate provided

We're constantly given, no matter the cert provided, 'unabel to verify HTTPS certificate chain: certificate not signed by known authority'

I'm assuming this is as Origin CA is not a trusted authority, is there any possible way to make Teleport accept CF Origin CA as a valid authority without running it without the insecure param?

Thanks in advance

Edit:
Just checked the deployment again, looks like someone has already added --insecure, but the issue still remains. Any clues?

**Comment by dudeisbrendan03** [comment]
@webvictim Sorry for the ping but you've usually had insight on issues that other people haven't provided, I've had 3 people look into this clueless and I still have no idea what's going on

**Reply by webvictim** [reply]
When is it that you're getting the error? When trying to connect to the web UI, or add an agent etc?

As an aside, Teleport validates certificate chains using standard Golang logic (which is to use the system trust store i.e. `/etc/ssl/certificates` or similar). Theoretically you should be able to add the Cloudflare Origin CA to the system trust store to make this go away.

**Reply by dudeisbrendan03** [reply]
I'll update the images to include the certificate in the system store and see how it goes!
It's when trying to deploy Teleport for the auth server + web UI

**Comment by evrynet1** [comment]
If you remove the universal SSL from CF then enable it again then the new certificate will be issued by LetsEncrypt rather than CloudFlare CA which might fixes the problem here.

**Comment by dudeisbrendan03** [comment]
We don't use Universal SSL, and this we're not even at that stage yet this
is just trying to start Teleport- trying to figure out if there's a better
method than putting it into the system cert store (since it's in
kubernetes, it's kind of a hassle to do, and I feel like it's stupid
mounting it as an opaque or something).

**Reply by webvictim** [reply]
There's a PR here (https://github.com/gravitational/teleport/pull/8372) that would add support for setting `SSL_CERT_DIR` in the `teleport-kube-agent` chart - if you raised a similar PR for the `teleport-cluster` chart we'd be happy to take a look.
